### PR TITLE
kmsv2: test cleanup

### PR DESCRIPTION
**Pull request by ritazh**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Cleanup few things in cmd/kube-apiserver/app/testing/testserver.go and staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go [hunk 1]
```diff
@@ -84,8 +82,8 @@ func WaitForBase64PluginToBeUp(plugin *Base64Plugin) error {
 		return gRPCErr == nil && resp.Healthz == "ok", nil
 	})
 
-	if pollErr == wait.ErrWaitTimeout {
-		return fmt.Errorf("failed to start kms-plugin, error: %v", gRPCErr)
+	if pollErr != nil {
```
Comment (aramase) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go:85]: Let's make this change in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go as well!

**Review by aramase — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go (hunk 1 above)
Comment (ritazh) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go:85]: I think you meant v1beta1. done!

File: cmd/kube-apiserver/app/testing/testserver.go
Comment (enj) [inline cmd/kube-apiserver/app/testing/testserver.go:1]: I would revert the changes to this file - this test server isn't guarantee to exit cleanly, and we do not want to start causing test flakes.

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go [hunk 2]
```diff
@@ -135,10 +133,8 @@ func (s *Base64Plugin) Start() error {
 // CleanUp stops gRPC server and the underlying listener.
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go:133]: Can you fix all the `Start()` call sites that spin up new go routine and fail to check the error?

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go:1]: A more through change would be to update `NewBase64Plugin` to take a `testing.T` as input, and then make the `Start`, `Cleanup`, and `WaitForBase64PluginToBeUp` methods private.  Inside of `NewBase64Plugin` you would call the new `start` and `waitForBase64PluginToBeUp` methods and do a `t.Cleaup(cleanup)`.  This would remove a lot of boilerplate from our tests.

**Review by enj — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go [hunk 3]
```diff
@@ -135,10 +142,8 @@ func (s *Base64Plugin) Start() error {
...
 func (s *Base64Plugin) CleanUp() {
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go:143]: Can we unexport this since manual clean up should never be required now?

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go [hunk 4]
```diff
@@ -135,10 +142,8 @@ func (s *Base64Plugin) Start() error {
...
 	s.grpcServer.Stop()
-	s.listener.Close()
-	if !strings.HasPrefix(s.socketPath, "@") || runtime.GOOS != "linux" {
-		os.Remove(s.socketPath)
-	}
+	_ = s.listener.Close()
+	_ = os.Remove(s.socketPath)
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go:146]: Do we need the same changes for the beta plugin?

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 5]
```diff
@@ -228,10 +207,10 @@ func TestIntermittentConnectionLoss(t *testing.T) {
 		t.Fatalf("failed when execute encrypt, error: %v", err)
 	}
 	t.Log("Connected to KMSPlugin")
+	f.CleanUp()
```
[the remark this answers is not in the document]
Comment (enj) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:210]: I would expect `CleanUp` to be unexported since manual cleanup shouldn't be required?

**Review by enj — COMMENTED** [review]
Minor comments.

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (ritazh) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:210]: It's needed https://github.com/kubernetes/kubernetes/blob/7ebe48f03c60fb49f155e018c712c5e564b50433/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go#L210-L214 because the test is simulating connection loss and a plugin stop and restart here https://github.com/kubernetes/kubernetes/blob/7ebe48f03c60fb49f155e018c712c5e564b50433/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go#L232

**Review by aramase — COMMENTED** [review]
/lgtm

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (enj) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:210]: Okay maybe in a follow up we should add a similar test for KMS v2?

[the remark this answers is not in the document]
Comment (ritazh) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:210]: https://github.com/kubernetes/kubernetes/issues/116999

[the remark this answers is not in the document]
Comment (aramase) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:210]: we have [TestIntermittentConnectionLoss](https://github.com/kubernetes/kubernetes/blob/release-1.27/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/grpc_service_unix_test.go#L201) test for KMSv2.
